### PR TITLE
Revamping GitHub Issue workflow

### PR DESCRIPTION
**Pull request by apurva-koti**
## What changes are proposed in this pull request?
 
This pull request addresses the due improvements to the issue workflow we have been discussing. It adds a few strict issue templates and adds an issue policy.
 
## How is this patch tested?
 
N/A
 
## Release Notes
We have updated our GitHub issue policy. Please refer to the policy before filing an issue. 

### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
Our GitHub issue policy has been updated to streamline the review process and ensure issues are addressed appropriately.
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -53,9 +53,11 @@ Official documentation for MLflow can be found at https://mlflow.org/docs/latest
 
 Community
 ---------
-To discuss MLflow or get help, please subscribe to our mailing list (mlflow-users@googlegroups.com) or
+To discuss MLflow, please subscribe to our mailing list (mlflow-users@googlegroups.com) or
```
Comment (smurching) [inline README.rst]: Could we list more specific channels of communication first & make the mailing list the fall-through, e.g.:

```
For help or questions about MLflow usage (e.g. "how do I do X?") see the `docs <https://mlflow.org/docs/latest/index.html>`_  or `Stack Overflow <https://stackoverflow.com/questions/tagged/mlflow>`_

To report a bug, file a documentation issue, or file a feature request, please open a GitHub issue.

For release announcements and other discussions, please subscribe to our mailing list (mlflow-users@googlegroups.com) or join us on Slack at https://tinyurl.com/mlflow-slack.
```

File: .github/ISSUE_TEMPLATE/bug_report_template.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+name: Bug Report
+about: Use this template for reporting bugs encountered while using MLflow.
+labels: 'bug'
+title: [BUG]
+---
+Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)
```
Comment (smurching) [inline .github/ISSUE_TEMPLATE/bug_report_template.md]: For this bug report template, be good to clarify here that users should ask for help debugging their code on SO rather than issues (in case people don't click through to the issue policy)

File: .github/ISSUE_TEMPLATE/bug_report_template.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  
+Please do not delete this template unless you are sure your issue is outside its scope.
+
+### System information
+- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
+- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
+- **MLflow installed from (source or binary)**: 
+- **MLflow version (run ``mlflow --version``)**:
+- **Python version**: 
+- **npm version** (if running the dev UI):
```
Comment (smurching) [inline .github/ISSUE_TEMPLATE/bug_report_template.md]:
```suggestion
- **npm version** (if running the dev UI - leave blank otherwise):
```

File: .github/ISSUE_TEMPLATE/feature_request_template.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  
+Please do not delete this template unless you are sure your issue is outside its scope.
+
+### Background information
+- **MLflow version (run ``mlflow --version``)**:
+- **Python version**: 
+- **Am I willing to create a pull request if approved?**
+
+### Describe the proposal
+Describe the problem clearly here. What is the use case in mind?
```
Comment (smurching) [inline .github/ISSUE_TEMPLATE/feature_request_template.md]:
```suggestion

Provide a clear high-level description of the feature request in the following sections. Feature requests that are likely to be accepted:
* Are minimal in scope (note that it's always easier to add additional functionality later than remove functionality)
* Are extensible (e.g. if adding an integration with an ML framework, is it possible to add similar integrations with other frameworks)
* Have user impact & value that justifies the maintenance burden of supporting the feature moving forwards. The [JQuery contributor guide](https://contribute.jquery.org/open-source/#contributing-something-new) has an excellent discussion on this.

### Motivation
What is the use case in mind?  Why is it valuable to support, and why is it currently difficult or impossible to achieve? Could the desired functionality alternatively be implemented as a third-party package using MLflow public APIs? 

### Proposed Changes
For user-facing changes, what APIs are you proposing to add or modify? For internal changes, what code paths will need to be modified? 

```

Also, I wonder if we can make this more detailed to encourage users to provide more info - see the [TensorFlow RFC template](https://github.com/tensorflow/community/blob/master/rfcs/yyyymmdd-rfc-template.md) for example. In particular, IMO it'd be nice to guide users towards a workflow of:

1) Submit feature request with high-level description
2) Discuss feature request with a committer, who may ask for a more detailed design 
3) After discussion & agreement on feature request, start implementation

Maybe we should just explicitly call out ^ as the recommended workflow/lifecycle for a feature request, and encourage users to describe their designs in detail (step 2 above) using the format in the TF RFC (my suggestion above tries to incorporate some elements from the TF RFC)

File: .github/ISSUE_TEMPLATE/installation_issue_template.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+labels: 'bug'
+title: [SETUP-BUG]
+---
+Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)
+for information on what types of issues we address.
+  
+Please do not delete this template unless you are sure your issue is outside its scope.
+
+### System information
+- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
```
Comment (smurching) [inline .github/ISSUE_TEMPLATE/installation_issue_template.md]: I think we can remove the custom code question? Doesn't seem as relevant for installation issues

File: ISSUE_POLICY.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Before filing an issue, make sure to search for related issues and check if
```
Comment (smurching) [inline ISSUE_POLICY.md]: Link to issue search?

File: ISSUE_POLICY.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+they address yours. Our policy is that GitHub issues are
+for bug reports, documentation fixes, feature requests, 
+and installation issues. Please do not delete the issue template placeholder unless you are certain your
+issue is outside its scope.
+  
+For support (ex. "How do I do X?"), please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).
```
Comment (smurching) [inline ISSUE_POLICY.md]: Maybe move this up to the top (i.e. right after the "Before filing an issue" sentence) so people are more likely to see it.

**Review by smurching — COMMENTED** [review]
Awesome this is looking great! Had a couple of suggestions

File: .github/ISSUE_TEMPLATE/bug_report_template.md (hunk 3 above)
Comment (apurva-koti) [reply .github/ISSUE_TEMPLATE/bug_report_template.md]: I think this is implied - feels a bit too wordy.

File: .github/ISSUE_TEMPLATE/feature_request_template.md (hunk 4 above)
Comment (apurva-koti) [reply .github/ISSUE_TEMPLATE/feature_request_template.md]: Discussed offline

File: ISSUE_POLICY.md (hunk 7 above)
Comment (ankitmathur-db) [inline ISSUE_POLICY.md]: Maybe "file an issue at" instead of "refer to"?

File: .github/ISSUE_TEMPLATE/feature_request_template.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+labels: 'enhancement'
+title: "[FR]"
+---
+Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)
+for information on what types of issues we address.
+  
+Please do not delete this template unless you are sure your issue is outside its scope.
+
+-------
+##Guidelines
```
Comment (smurching) [inline .github/ISSUE_TEMPLATE/feature_request_template.md]:
```suggestion
## Guidelines
```

for formatting

File: .github/ISSUE_TEMPLATE/feature_request_template.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+Feature requests should involve the following workflow:
```
Comment (smurching) [inline .github/ISSUE_TEMPLATE/feature_request_template.md]:
```suggestion
Feature requests typically go through the following lifecycle:
```

File: .github/ISSUE_TEMPLATE/feature_request_template.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+1. Submit feature request with high-level description
```
Comment (smurching) [inline .github/ISSUE_TEMPLATE/feature_request_template.md]:
```suggestion
1. Submit feature request with high-level description on GitHub issues (this is what you're doing now)
```

**Review by smurching — COMMENTED** [review]
Two small suggestions, otherwise LGTM!
